constantine/math/config/precompute.nim

```diff
@@ -392,6 +392,16 @@ func primePlus1div2*(P: BigInt): BigInt =
   let carry = result.add(1)
   doAssert not carry
 
+func primeMinus1div2*(P: BigInt): BigInt =
+  ## Compute (P-1)/2
+  ## For use in constant-time modular inversion
+  ##
+  ## Warning ⚠️: Result is in the canonical domain (not Montgomery)
+
+  result = P
+  discard result.sub(1)
+  result.shiftRight(1)
```

Comment:
Nitpick: actually let's pick 6 and 7, in binary they are `0b110` and `0b111`, after a right shift they both become `0b11` which is 3, so the -1 is free for odd integers